src/pages/Subjects.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect, ReactNode } from "react";
+import Layout from "../components/Layout/Layout";
+interface Subject {
+  [x: string]: ReactNode;
+  id: number;
+  name: string;
+  description: string;
+}
+
+const Subjects: React.FC = () => {
+  const [subjects, setSubjects] = useState<Subject[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState("");
+
+  const fetchSubjects = async () => {
```

Comment:
Does this need to be async (): Promise<void>  for the return type?

Reply:
'fetch' is asynchronous so returns a promise that has to be processed using 'await', right?

Reply:
Inserting a return type is void is just declaring that the function doesn't return anything so it's just for consistency and making use of more type protection. Strictly speaking its not needed since its inferred but I guess it depends on how far you want to be strict with yourself.